src/hevm/src/EVM.hs

```diff
@@ -1535,20 +1537,20 @@ askSMT codeloc condition continue = do
      Just w -> choosePath (Case w)
      -- If this is a new query, run the query, cache the result
      -- increment the iterations and select appropriate path
-     Nothing -> do pathconds <- use pathConditions
+     Nothing -> do pathconds <- use constraints
                    assign result . Just . VMFailure . Query $ PleaseAskSMT
-                     condition pathconds choosePath
+                     condition (fst <$> pathconds) choosePath
 
    where -- Only one path is possible
          choosePath :: BranchCondition -> EVM ()
          choosePath (Case v) = do assign result Nothing
-                                  pushTo pathConditions (if v then condition else sNot condition)
+                                  pushTo constraints (if v then (condition, UnOp "not" whiff) else (sNot condition, whiff))
```

Comment:
should the `not` be in the else branch here?

Reply:
yeah, I think so